handler/src/main/java/io/netty/handler/ssl/ApplicationProtocolNegotiationHandler.java

```diff
@@ -168,7 +178,7 @@ private void removeSelfIfPresent(ChannelHandlerContext ctx) {
     /**
      * Invoked on failed initial SSL/TLS handshake.
      */
-    protected void handshakeFailure(ChannelHandlerContext ctx, Throwable cause) throws Exception {
+    protected void handshakeFailure(ChannelHandlerContext ctx, Throwable cause) {
```

Comment:
@NiteshKant this is a breaking change as the method is protected and a user may have overridden it.

Reply:
I fixed this and pushed to your branch.